TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -16,10 +18,120 @@ public static class SavepointStatementChecker
                 if (savePointName.StartsWith("SYS", StringComparison.OrdinalIgnoreCase))
                 {
                     DiagnosticUtils.AddError(savepointStatement,
-                        "Invalid savepoint-name, it must not begin with 'SYS'.",
+                        "Invalid savepoint-name, it must not begin with 'SYS'.", context);
+                }
+            }
+        }
+    }
+
+    internal static class AlterSequenceStatementChecker
+    {
+        public static void OnCodeElement(AlterSequenceStatement alterSequenceStatement,
+            List<string> duplicatedClauses, bool emptyClauseSet,
+            CodeElementsParser.AlterSequenceStatementContext context)
+        {
+            foreach (var clauseType in duplicatedClauses)
+            {
+                DiagnosticUtils.AddError(alterSequenceStatement,
+                    clauseType + " cannot be specified more than once",
+                    context);
+            }
+
+            if (emptyClauseSet)
+            {
+                DiagnosticUtils.AddError(alterSequenceStatement,
+                    "At least one option among RESTART, INCREMENT, MINVALUE, MAXVALUE, CYCLE, CACHE or ORDER must be specified",
+                    context);
+            }
+
+            if (alterSequenceStatement.MinValue != null)
+            {
+                var tokenValue = alterSequenceStatement.MinValue.Literal.LiteralValue;
+                if (tokenValue.Type != LiteralTokenValueType.Decimal) return;
+                var decimalLiteral = (DecimalLiteralTokenValue) tokenValue;
+                bool isInteger = Math.Abs(decimalLiteral.Number % 1) < double.Epsilon;
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
```

Comment:
The IBM error is `THE SCALE OF THE DECIMAL NUMBER MUST BE ZERO`.
I'd prefer `Only digits '0' are allowed after decimal point`.

This apply to other `decimal should be without non-zero digits in the right of the decimal point` messages.